[radiobrowser] Initial contribution (#16392)
openhab/openhab-addons: bundles/org.openhab.binding.radiobrowser/README.md

# Radio Browser Binding

Radio Browser is a community driven database of internet radio and TV stations, that has an open API that is free for all to use and has multiple servers to ensure up time is high.
With this binding you can use their database of more than 45,000 stations to apply filters and quickly find internet radio streams that interest you.

## Supported Things

- `radio`: Add one of these manually, and it should come online after fetching language and country data to populate the filter channels.

### `radio` Thing Configuration

| Name          | Type    | Description                                                                                 | Default | Required | Advanced |
|---------------|---------|---------------------------------------------------------------------------------------------|---------|----------|----------|

> | filters       | text    | Allows you to specify your own filters from the `advanced search` of the API.           | `?hidebroken=true&limit=1700&order=clickcount`                                                                          | yes      | no      |

Reviewer: Can you fix the table markup?

Author: done

| clicks        | boolean | Helps to support the server recommend good results.                                         | true    | yes      | yes      |
| languageCount | integer | If you want less languages to be shown as a filter, you can raise this or create your own.  | 14      | yes      | yes      |

## Filters Configuration

> Found by ticking the `show advanced` box, this configuration called `filters` can be used to limit the stations based on codecs, minimum quality, ordering and more.

Reviewer: ```suggestion
The `advanced` configuration parameter `filters` can be used to limit the stations based on different fields like codecs, minimum quality, ordering and more.
All possible filter options are listed here <https://de1.api.radio-browser.info/#Advanced_station_search>
```

> The default filter is this and can be change to create better results for your needs.

Reviewer: ```suggestion
The default filter (below) can be changed to suit your needs.
```

> ```

Author: I do not understand, so will wait for further clarification on this.

> ```

Reviewer: Discussion: https://github.com/openhab/openhab-docs/issues/999
```suggestion
```text
```

> ?hidebroken=true&limit=1700&order=clickcount

Reviewer: I know it is an advanced channel, but for users it would also be simplify this configuration like `key1=value1,key2=value2` and validate/construct an uri parameter string from that.

Author: I agree it would make it slightly more simple so have done it, but do not believe it was worth the work changing it and then redoing documentation that is already written for the small gain.

```

To give an example on how to show less stations by removing any low quality streams below 128kbps (FM radio quality), this could be added.
You may not want to do this if you're interested in chat and sports channels.

```
?hidebroken=true&limit=1700&order=clickcount&bitrateMin=128
```

> 
> If you want to create your own custom filter, the options are listed here <https://de1.api.radio-browser.info/#Advanced_station_search>

Reviewer: ```suggestion
```

Examples on how to do searches from rules, or you can also change an item to take input by using `oh-input-item` using metadata called `Default list item widget`.

Search for all stations that contain `hit` in their name, and auto select the first result.

```
Radio_Station.sendCommand("hit")
```

Search and auto select the station if you know the UUID from the website.

```
Radio_Station.sendCommand("b6a490e8-f498-4a7c-b024-607b3d997614")
```

Clear any manual search results using the above two methods, and `REFRESH` back to using the normal filter channels.

```
Radio_Station.sendCommand(REFRESH)
```

## Show Your Thanks

If you enjoy the binding, then please consider sponsoring or giving a once off tip as a thank you via the links.
This allows me to purchase software and hardware to contribute more bindings, and also shows this binding should gain more feature like favourites.
Some coffee to keep me coding faster never hurts :slight_smile:

Sponsor @Skinah on GitHub
<https://github.com/sponsors/Skinah/>

Paypal can also be used via

> matt A-T pcmus D-O-T C-O-M

Reviewer: While i wish you a lot of donations and coffee, I'm not sure if this is allowed to be added to the official documentation of the openhab website. Need some advice from other @openhab/add-ons-maintainers if this is oke or not.

Author: Have removed this from the readme and will instead reply on the widget to make this suggestion.
